codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpBoundProtocolGenerator.kt

```diff
@@ -159,6 +159,22 @@ private class ServerHttpBoundProtocolTraitImplGenerator(
         val operationName = symbolProvider.toSymbol(operationShape).name
         val inputName = "${operationName}${ServerHttpBoundProtocolGenerator.OPERATION_INPUT_WRAPPER_SUFFIX}"
 
+        val impliedResponseContentType = httpBindingResolver.responseContentType(operationShape)
+        val verifyResponseContentType = if (impliedResponseContentType == null) { "" } else {
+            """
+            if let Some(headers) = req.headers() {
+                if let Some(accept) = headers.get(#{http}::header::ACCEPT) {
+                    if accept != ${impliedResponseContentType.dq()} {
```

Comment:
```suggestion
                    if accept != "${impliedResponseContentType}" {
```

`.dq` is useful when you're interpolating within single double quotes (`""`).